libafl/src/fuzzer/replaying.rs

```diff
@@ -0,0 +1,852 @@
+//! Fuzzer instance that increases stability by executing the same input multiple times.
+
+use alloc::{borrow::Cow, string::ToString, vec::Vec};
+use core::{fmt::Debug, hash::Hash, marker::PhantomData};
+
+use hashbrown::HashMap;
+use libafl_bolts::{
+    current_time, generic_hash_std,
+    tuples::{Handle, MatchName, MatchNameRef},
+};
+use serde::Serialize;
+
+#[cfg(feature = "std")]
+use crate::fuzzer::BloomInputFilter;
+#[cfg(feature = "introspection")]
+use crate::monitors::PerfFeature;
+use crate::{
+    corpus::{Corpus, CorpusId, HasCurrentCorpusId, HasTestcase, Testcase},
+    events::{
+        CanSerializeObserver, Event, EventConfig, EventFirer, EventProcessor, ProgressReporter,
+    },
+    executors::{Executor, ExitKind, HasObservers},
+    feedbacks::Feedback,
+    fuzzer::{
+        Evaluator, EvaluatorObservers, ExecuteInputResult, ExecutesInput, ExecutionProcessor,
+        Fuzzer, HasFeedback, HasObjective, HasScheduler, InputFilter, NopInputFilter,
+        STATS_TIMEOUT_DEFAULT,
+    },
+    inputs::Input,
+    mark_feature_time,
+    monitors::{AggregatorOps, UserStats, UserStatsValue},
+    observers::ObserversTuple,
+    schedulers::Scheduler,
+    stages::{HasCurrentStageId, StagesTuple},
+    start_timer,
+    state::{
+        HasCorpus, HasCurrentTestcase, HasExecutions, HasLastFoundTime, HasLastReportTime,
+        HasSolutions, MaybeHasClientPerfMonitor, Stoppable,
+    },
+    Error, HasMetadata,
+};
+
+/// A fuzzer instance for unstable targets that increases stability by executing the same input multiple times.
+///
+/// The input will be executed as often as necessary until the most frequent result appears at least `min_count_diff` times more often than any other result and at most `max_trys` times.
+#[derive(Debug)]
+pub struct ReplayingFuzzer<CS, F, O, IF, OF> {
+    min_count_diff: usize,
+    max_trys: usize,
+    handle: Handle<O>,
+    scheduler: CS,
+    feedback: F,
+    objective: OF,
+    input_filter: IF,
+}
+
+impl<CS, F, O, IF, OF, S> HasScheduler<<S::Corpus as Corpus>::Input, S>
```

Comment:
we stopped to use 
`<S::Corpus as Corpus>::Input`
after https://github.com/AFLplusplus/LibAFL/pull/2854

can you add I to your generic for representing the input type